test/cmocka/src/lib/lib/bzero.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2018, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Michal Jerzy Wierzbicki <michalx.wierzbicki@linux.intel.com>
+ */
+
+#include <sof/alloc.h>
+
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+static int checkArr(char *arr, size_t n, char shouldBe)
+{
+	size_t i;
+
+	for (i = 0; i < n; ++i)
+		if (arr[i] != shouldBe)
+			return i;
+	return -1;
+}
+
+static void test_lib_lib_bzero_checkArr(void **state)
+{
+	char str[] = "aaaaaa";
+	const int size = sizeof(str)-1; // last char is /0
+
+	(void) state;
+	assert_int_equal(checkArr(str, size, 'a'), -1);
+}
+
+static void test_lib_lib_bzero_char_zero_none(void **state)
+{
+	char str[] = "aaaaaa";
+	const int size = sizeof(str)-1;
+
+	(void) state;
+	bzero(str, 0);
+	assert_int_equal(checkArr(str, size, 'a'), -1);
+}
+
+static void test_lib_lib_bzero_char_zero_all(void **state)
+{
+	char str[] = "aaaaaa";
+	const int size = sizeof(str)-1;
+
+	(void) state;
+	bzero(str, size);
+	assert_int_equal(checkArr(str, size, 0), -1);
+}
+
+static void test_lib_lib_bzero_char_zero_middle(void **state)
+{
+	//            aa000aa - pattern of zeroing
+	char str[] = "aaaaaaa";
+	const int size = sizeof(str)-1;
+	const int zeroBegin = 2;
+	const int zeroEnd = 4;
+
+	(void) state;
+	bzero(str+zeroBegin, zeroEnd-zeroBegin);
+	assert_int_equal(checkArr(
```

Comment:
Please, can you split it into 3 unit tests so we have 1 assert per test?

Reply:
Ok, will split into 3 uts with same data/logic but different checks.